src/sinks/util/encoding.rs

```diff
@@ -156,7 +159,23 @@ pub trait EncodingConfiguration<E> {
                 Event::Log(log_event) => {
                     let to_remove = log_event
                         .keys()
-                        .filter(|f| !only_fields.contains(f))
+                        .filter(|field| {
+                            let field_path = PathIter::new(field).collect::<Vec<_>>();
+                            !only_fields.iter().any(|only| {
+                                if only == field {
+                                    true // fastpath
+                                } else {
+                                    // TODO: This is a hack for #2407, #2410 should fix this fully.
+                                    // This is *slow*.
+                                    let only_path = PathIter::new(only);
+                                    if field_path.starts_with(&only_path.collect::<Vec<_>>()[..]) {
+                                        true
+                                    } else {
```

Comment:
Don’t we need to parse the path for the except_fields loop below as well?

Reply:
I checked just to be safe but we don't. I added the tests to this PR though.

![image](https://user-images.githubusercontent.com/130903/80045731-3456a380-84bd-11ea-865f-015ccd09198e.png)